app/models/alchemy/page/page_natures.rb

```diff
@@ -20,7 +20,7 @@ def rootpage?
     end
 
     def systempage?
-      return true if Page.root.nil?
+      return true if Page.count.zero?
```

Comment:
This won't help unfortunately. The caching only works when the collection is already loaded and we reload it every time.